File: mpp/lab3/src/main/java/domain/Score.java
package domain;

import java.util.HashMap;
import java.util.Map;

public class Score {
    private final Participant participant;
    private Map<ScoreType, Integer> scores = new HashMap<>();

    public Score(Participant participant) {
        this.participant = participant;
    }

    public Participant getParticipant() {
        return participant;
    }

    public int getScore(ScoreType type) {
        return scores.getOrDefault(type, 0);
    }

    public void setScore(ScoreType type, int score) {
        scores.put(type, score);
    }

    public void setScores(Map<ScoreType, Integer> scores) {
        this.scores = scores;
    }

    public Map<ScoreType, Integer> getScores() {
        return scores;
    }

    public int getTotalScore() {
        return scores.values().stream().reduce(0, Integer::sum);
    }

    @Override
    public String toString() {
        return "ParticipantScore{" +
                "participant=" + participant +
                ", totalScore=" + getTotalScore() +
                '}';
    }
}

File: mpp/lab7/server/Domain/Score.cs
using System.Collections.Generic;
using System.Linq;
using Triathlon;

namespace server.Domain
{
    public class Score
    {
        public Participant Participant { get; set; }
        public Dictionary<ScoreType, int> Scores { get; set; } = new Dictionary<ScoreType, int>();

        public Score(Participant participant)
        {
            Participant = participant;
        }

        public int GetScore(ScoreType type)
        {
            Scores.TryGetValue(type, out var value);
            return value;
        }

        public void SetScore(ScoreType type, int value)
        {
            Scores[type] = value;
        }

        public int GetTotalScore()
        {
            return Scores.Sum(score => score.Value);
        }

        public override string ToString()
        {
            return $"Score{{Participant={Participant}, TotalScore={GetTotalScore()}}}";
        }

        public ScoreProto ToProto()
        {
            var protoScores = new Dictionary<int, int>();
            foreach (var (type, value) in Scores)
            {
                protoScores.Add((int) type, value);
            }
            
            return new ScoreProto { Participant = Participant.ToProto(), Scores = { protoScores }};
        }
    }
}
